packages/provider-elasticsearch/src/example-types/results/highlightUtil.test.js

```diff
@@ -0,0 +1,55 @@
+import { mergeHitHighlights } from './highlightUtil.js'
+
+let nodeHighlight = {
+  pre_tags: ['<em>'],
+  post_tags: ['</em>'],
+}
+
+describe('mergeHitHighlights()', () => {
```

Comment:
Another test case we should add is when there are identical substrings being highlighted more than once. Ex: `'<em>lazy</em> for was really <em>lazy</em>`